server/cluster_worker.go

```diff
@@ -295,8 +299,8 @@ func (c *raftCluster) handleAddPeerReq(region *metapb.Region) (uint64, error) {
 		storeID := store.GetId()
 
 		existStore := false
-		for _, peerStoreID := range region.StoreIds {
-			if peerStoreID == storeID {
+		for _, peer := range region.Peers {
+			if peer.GetStoreId() == storeID {
 				// we can't add peer in the same store.
```

Comment:
tip: maybe we'll allow this later, but allocate a new peer id and recycle the old peer.